### PR TITLE
feat(datasets): move default mode of ManagedTableDataSet to read-only

### PR DESCRIPTION
**Pull request by PetitLepton**
## Description
By default, `ManagedTableDataSet` has `overwrite` as `write_mode`. This may be a bit dangerous as a user may not be aware that the `save` method would wipe out the existing data. In this PR, I suggest to make the default behavior as read-only for the user to clearly specify the expected write behavior. 

## Development notes
I chose to switch `write_mode` to `None` as the default and raise an exception if the user tries to save the data set without specifying the mode. The tests have been updated to reflect this behavior. Tests which rely on a default `write_mode` being `overwrite` have been updated with the specified `write_mode`.

This is a breaking change as `write_mode` is None preventing `save` by default

## Checklist

- [X] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the relevant `RELEASE.md` file
- [X] Added tests to cover my changes

### CODE REVIEW

File: kedro-datasets/kedro_datasets/databricks/managed_table_dataset.py [hunk 1]
```diff
@@ -365,6 +368,11 @@ def _save(self, data: Union[DataFrame, pd.DataFrame]) -> None:
         Args:
             data (Any): Spark or pandas dataframe to save to the table location
         """
+        if self._table.write_mode is None:
+            raise DataSetError(
+                "'save' can not be used in read-only mode. "
+                "Change 'write_mode' to overwrite, upsert or append."
```
Comment (datajoely) [inline kedro-datasets/kedro_datasets/databricks/managed_table_dataset.py:374]:
```suggestion
                "Change 'write_mode' value to `overwrite`, `upsert` or `append`."
```

**Review by datajoely — APPROVED** [review]
I think this is a great push @PetitLepton

**Review by noklam — COMMENTED** [review]
This looks good to me, would you update the `RELEASE.md` to reflect this change?

**Comment by PetitLepton** [comment]
> This looks good to me, would you update the `RELEASE.md` to reflect this change?

I added two sentences. Let me know if it is what you had in mind.

**Comment by noklam** [comment]
@PetitLepton thanks! Looks great.

**Review by noklam — APPROVED** [review]

**Comment by PetitLepton** [comment]
@noklam and @datajoely I updated the docstring to reflect the changes which should complete the list of requirements from the PR template. Anything more to do on my side?

PS: thanks a lot for your reviews and guidance.

**Comment by noklam** [comment]
@PetitLepton  The DCO is failing, would you be able to fix it? If you click this link there are [Details](https://github.com/kedro-org/kedro-plugins/pull/303/checks?check_run_id=15904730676) to fix this.

**Review by datajoely — APPROVED** [review]

**Review by ankatiyar — APPROVED** [review]
💯

**Pull request merged**